robots/gazebo_tank.h

```diff
@@ -0,0 +1,64 @@
+#pragma once
+
+// BoB robotics includes
+#include "../common/pose.h"
+#include "../common/stopwatch.h"
+#include "tank.h"
+
+// Third-party includes
+#include "../third_party/units.h"
+#include <utility> 
+
+// Gazebo includes
+#include <gazebo/transport/transport.hh>
+#include <gazebo/msgs/msgs.hh>
+
+namespace BoBRobotics {
+namespace Robots {
+using namespace units::literals;
+
+
+class GazeboTank
+  : public Tank
+{
+    using radians_per_second_t = units::angular_velocity::radians_per_second_t;
+public:
+    GazeboTank(const radians_per_second_t maximumSpeed, gazebo::transport::NodePtr node)
+      : m_MaximumSpeed(maximumSpeed)
+    {
+        // Publish to the  differential_drive_robot topic
+        pub =node->Advertise<gazebo::msgs::Vector3d>("~/differential_drive_robot/vel_cmd");;
+        // Wait for a subscriber to connect to this publisher
+        pub->WaitForConnection();
+    }
+
+    virtual meters_per_second_t getAbsoluteMaximumSpeed() const override
+    {
+        return 0.3* m_MaximumSpeed; //radius of the wheel is 0.3m. v = r × ω
```

Comment:
I'm getting a compiler error here. The problem is that the units library won't let you convert between different kinds of units here (even though the equation you're using is correct). You have to get a ``double`` out and construct a ``meters_per_second_t`` object, like so:
```c++
return meters_per_second_t{ 0.3 * m_MaximumSpeed.value() };
```